sklearn/utils/estimator_checks.py

```diff
@@ -163,7 +163,11 @@ def _yield_checks(estimator):
             # We skip pairwise because the data is not pairwise
             yield check_sample_weights_shape
             yield check_sample_weights_not_overwritten
-            yield check_sample_weight_equivalence
+            yield check_sample_weight_equivalence_on_dense_data
+            # FIXME: filter on tags.input_tags.sparse
+            # (estimator accepts sparse arrays)
+            # once issue #30139 is fixed.
+            yield check_sample_weight_equivalence_on_sparse_data
```

Comment:
shouldn't the check already be there? I'm not sure if I understand why we don't have it here.

Reply:
Not yet,  the PR #30187 fixing the `tags.input_tags.sparse` needs to be merged.